tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -290,25 +290,27 @@ private static void SetLegacyResourceNames(BeforeActionStruct typedArg, Span spa
                 routeTemplate = $"{controllerName}/{actionName}";
             }
 
-            string resourceName = $"{httpMethod} {routeTemplate}";
-
-            // override the parent's resource name with the MVC route template
-            span.ResourceName = resourceName;
+            return $"{httpMethod} {routeTemplate}";
         }
 
-        private static Span StartMvcCoreSpan(Tracer tracer, Span parentSpan, BeforeActionStruct typedArg, HttpContext httpContext, HttpRequest request)
+        private static Span StartMvcCoreSpan(
+            Tracer tracer,
+            AspNetCoreHttpRequestHandler.RequestTrackingFeature trackingFeature,
+            BeforeActionStruct typedArg,
+            HttpContext httpContext,
+            HttpRequest request)
         {
             // Create a child span for the MVC action
             var mvcSpanTags = new AspNetCoreMvcTags();
-            var mvcScope = tracer.StartActiveInternal(MvcOperationName, parentSpan.Context, tags: mvcSpanTags);
+            var mvcScope = tracer.StartActiveInternal(MvcOperationName, tags: mvcSpanTags);
```

Comment:
Not even sure why we used parentSpan.Context before, I guess for minor efficiencies, but it would be wrong to use the rootSpan.Context here

Reply:
If we assume the customer has a trace (so far) like this:

```
|----aspnetcore.request------------- (still open)
     |---- custom.middleware--------- (still open)
```

Then we want the `aspnetcore.request_mvc` request to be a child of the current open span (`custom.middleware`). If we used `rootSpan.Context` it would be a sibling of `custom.middleware`, which would give a weird trace.

Reply:
Ahhhhh yep that makes sense thanks!